### ISSUE TITLE
Synonymization/generality problems

### ISSUE THREAD

**Issue by dkoslicki** [NCATSTranslator/Tests#62]
For test 7 asset 79, the [ARAX result](https://arax.ncats.io/?r=01f93955-e69e-4f20-9ee0-d139b96a4e5a) has progesterone as result 2 of 103. But the asset asks for _progestin_ (the synthetic version of progesterone, iirc).

For test 12, asset 309, pegfilgrastim is asked for, but any filgrastim or G-CSF should be accepted.

For test 13, asset 362, the goal is to get `ACE inhibitors` and ARAX returns `Angiotensin-converting enzyme inhibitors` (i.e. the full name)

For test case 15, asset 318, `Ivacaftor` is asked for and ARAX returns `Ivacaftor / texacaftor` and `ivacaftor /lumacaftor` as results 2 and 3. Do we want to do the testing post-conflation?

Test 24, asset 396 `Butyrate derivatives` is asked for, but `butyrate` and `butyric acid` aren't accepted as answers.

Test 29, asset 504, `thrombin` is asked for, but `alpha-thrombin` is not accepted as an answer (similarly for test 30, asset 595)

Test 35 asset 665, `KRASG12D inhibitor MRTX1133` is asked for, but `MRTX1133` (CHEMBL.COMPOUND:CHEMBL5081048) is not accepted

**Comment by dkoslicki** [comment]
Similarly Test 9 asset 623, `Rabeprazole` is asked for, but `Rabeprazole sodium` is not accepted as correct. Again with the post/pre-conflation issue

**Comment by colleenXu** [comment]
Note this issue for Asset 362 https://github.com/NCATSTranslator/Tests/issues/76

**Comment by colleenXu** [comment]
We noted the same issues for some tests:
* asset 396 Butyrate Derivatives (UMLS:C0581820): BTE returned entities that seem related like butyric acid (CHEBI:30772), D-alpha-aminobutyric acid (CHEBI:28797), and phenylbutyrate (CHEBI:41500)

**Comment by colleenXu** [comment]
Here's some more tests with the same kind of problem (I'm closing their separate issues and moving the info here):
* asset 32 Lactase `CHEMBL.COMPOUND:CHEMBL2108505`:
    * the NodeNorm entry for this ID seems odd: it has no other equivalent IDs ([Dev NodeNorm](https://nodenormalization-sri.renci.org/get_normalized_nodes?curie=CHEMBL.COMPOUND:CHEMBL2108505&conflate=true&drug_chemical_conflate=true&description=false)). 
    * There are other lactase entities with more equivalent IDs that show up in BTE's results with higher scores/ranks: [MESH:D043322](https://nodenormalization-sri.renci.org/get_normalized_nodes?curie=MESH:D043322&conflate=true&drug_chemical_conflate=true&description=false) (ChemicalEntity) and [UMLS:C0083183](https://nodenormalization-sri.renci.org/get_normalized_nodes?curie=UMLS:C0083183&conflate=true&drug_chemical_conflate=true&description=false) (Protein)
    * I've [noted before](https://github.com/NCATSTranslator/Tests/issues/32) that tools tended to fail this test. I haven't done this comparison lately though

**Comment by maximusunc** [comment]
So I think this gets to a root question: The Test is asking for a superclass identifier (in most cases) and an ARA gives back a subclass (more specific). Is the ARA expected to also return the superclass? If not, why not?

**Comment by dkoslicki** [comment]
From the subclass reasoning perspective (if a general concept is asked for, also return more specific instances), my impression was that precision is preferred to generality. But I guess it hasn't really been codified one way or another. My $0.02 is that a subclass being should "count" for the test asking for a superclass. Open to other opinions though

**Comment by maximusunc** [comment]
I agree that more specific instances are preferred, but does that mean that the general concept that was asked for should not be returned?

If we were to support subclass answers in the tests, how exactly would that work? Is there an easy way to determine if something is a subclass of the expected output?

**Comment by maximusunc** [comment]
This general topic will be discussed during the testing session of the upcoming Relay.

**Comment by sandrine-muller-research** [comment]
wrt "wrong output IDs": I guess not all tests are from me but the purpose of the test assets is to test the UI so when the UI returns something wrong (e.g. superclasses) the ID given by the UI is reported. 

wrt synonymization/generalization: 
- since the testing is testing the UI, the testing is done "post conflation" if the creative query returned by ARA does the conflation. At least that's how I made the tests.
- When a super set family was outputed the question I was asking myself was "are all BioLink entities in that superclass" passing/failing. If not then it is "too general" and is a failure. 
- for the thrombin case: I have not idea why alpha-thrombin is not accepted because I did not write a fail test for that one.

**Comment by colleenXu** [comment]
[Asset 79](https://github.com/NCATSTranslator/Tests/blob/main/test_assets/Asset_79.json) also has this issue. It's been running in sprints 5-6. 
The answer is "progestin" `MESH:D011372`, but this is a drug class with ~ 60 individual chemicals. Some of those individual chemicals (including progesterone itself) are high-ranking ARA results and should probably pass...

No one has been passing this test is sprints 5-6 (also noted [previously](https://github.com/NCATSTranslator/Tests/issues/32)). 

This test is also flagged in https://github.com/NCATSTranslator/Tests/issues/93
